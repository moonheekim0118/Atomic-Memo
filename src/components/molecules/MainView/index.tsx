import React from 'react';
import Span from '../../atoms/Span';
import styled from 'styled-components';

interface Props {
  text: string;
}

const MainView = ({ text }: Props) => {
  return (
    <StyledMainView>
      <Span size="normal" display="block">
        {text}
      </Span>
    </StyledMainView>
  );
};

const StyledMainView = styled.div`
  position: relative;
  display: flex;
  flex-direction: column;
  min-height: 100%;
  padding: 2em;
`;

export default MainView;
