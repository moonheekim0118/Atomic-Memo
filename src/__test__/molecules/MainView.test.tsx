import 'jsdom-global/register';
import React from 'react';
import MainView from '../../components/molecules/MainView';
import { mount } from 'enzyme';

describe('<MainView/>', () => {
  let container = null;

  it('renders correctly', () => {
    container = mount(<MainView text="test" />);
  });

  it('matches snpashot', () => {
    expect(container.html()).toMatchSnapshot();
  });
});
