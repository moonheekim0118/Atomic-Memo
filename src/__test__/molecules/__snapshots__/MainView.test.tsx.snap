// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainView/> matches snpashot 1`] = `"<div class=\\"MainView__StyledMainView-sc-1f2rj8k-0 iyqmxW\\"><span width=\\"auto\\" display=\\"block\\" class=\\"Span__StyledSpan-sc-10y5m6u-0 ftDQbL normal\\">test</span></div>"`;
